taskcluster/rb_taskgraph/signing.py

```diff
@@ -17,11 +17,10 @@
 
 
 @transforms.add
-def build_signing_task(config, tasks):
+def define_signing_flags(config, tasks):
     for task in tasks:
-        dep = task.pop("primary-dependency")
-        task["dependencies"] = {"build": dep.label}
-        task["name"] = dep.label[len(dep.kind) + 1 :]
+        dep = task["primary-dependency"]
+        task["name"] = dep.label[len(dep.kind) + 1:]
```

Comment:
It could be worth documenting what this operation does, or how `dep.label` is normally formatted that allows you to extract the name with this technique